spark/sql/utils/check.go

```diff
@@ -0,0 +1,25 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import "fmt"
+
+func WarnOnError(f func() error) {
+	if err := f(); err != nil {
+		fmt.Println("Received error:", err)
```

Comment:
You should not use fmt.Println from within library code. At the very minimum, use a logger but do consider a slightly more general version where you can pass in an error handler (which can then in turn be something that logs

Reply:
Done, added a callback.